### PR TITLE
ARROW-6266: [Java] Resolve the ambiguous method overload in RangeEqualsVisitor

### PR DESCRIPTION
**Pull request by liyafan82**
In RangeEqualsVisitor, there are overload methods for both super class and sub class. This will lead to unexpected behavior.

For example, if we call RangeEqualsVisitor#visit(v), where v is a fixed width vector, the method actually called may be visit(ValueVector), which is unexpected.

In general, in the visitor pattern, it is not a good idea to support method overload for both super class and sub-class as parameters.

### CODE REVIEW

**Comment by liyafan82** [comment]
cc @tianchen92

**Review by tianchen92 — APPROVED** [review]
Thanks @liyafan82 , LGTM.

File: java/vector/src/main/java/org/apache/arrow/vector/ExtensionTypeVector.java [hunk 1]
```diff
@@ -260,6 +260,6 @@ public BufferAllocator getAllocator() {
 
   @Override
   public boolean accept(RangeEqualsVisitor visitor) {
-    return visitor.visit(getUnderlyingVector());
+    return visitor.visitGeneral(getUnderlyingVector());
```
Comment (pravindra) [inline java/vector/src/main/java/org/apache/arrow/vector/ExtensionTypeVector.java]: can we instead do -

return getUnderlyingVector().accept(visitor)

Comment (tianchen92) [reply java/vector/src/main/java/org/apache/arrow/vector/ExtensionTypeVector.java]: I think this is also ok, and then should remove RangeEqualsVisitor#visit(ValueVector left, Void value).

Comment (tianchen92) [reply java/vector/src/main/java/org/apache/arrow/vector/ExtensionTypeVector.java]: Sorry for correction, if we don't have visitGeneral I think the API below will throw exception

>public boolean equals(ValueVector left) {
    return this.visit(left, null);
  }

Comment (liyafan82) [reply java/vector/src/main/java/org/apache/arrow/vector/ExtensionTypeVector.java]: The method suggested by @pravindra works. With the suggested change, method visit(ValueVector left) can be removed altogether.

**Review by pravindra — APPROVED** [review]

**Pull request closed without merging**